homeassistant/components/mill/climate.py

```diff
@@ -119,15 +121,20 @@ def __init__(self, coordinator, heater):
             )
         else:
             self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+            self._attr_target_temperature_step = PRECISION_HALVES
 
         self._update_attr(heater)
 
     async def async_set_temperature(self, **kwargs):
         """Set new target temperature."""
         if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
             return
+        if self._generation == 3:
+            temperature = float(temperature)
```

Comment:
I'm not sure if this is needed. Can I use float for all of the generations? As I said in the description, I don't have a first or second gen heater, so I can't verify. But I belive the API supports `float` for all generations, but is rouded down for < 3. gen.

Reply:
If you set the precision correctly I think you can use the temperature helper https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/temperature.py#L11